### PR TITLE
fix(@aws-amplify/ui-components): Check for 'amplify-redirected-from-hosted-ui' instead of 'amplify-signin-with-hostedUI'

### PR DESCRIPTION
**Pull request by ericclemmons**
**EDIT: Fixed by mirroring #4258 by @Amplifiyer**.

While working on https://github.com/aws-amplify/amplify-js/pull/5676, I discovered that refreshing the page in Firefox after being redirected back from Cognito Hosted UI gets stuck on _Loading..._.

The UI was coupled to an internal storage token `amplify-signin-with-hostedUI`) to bypass checking for the current authenticated user. (This only happens once, because the UI removes this token immediately afterwards)

This PR removes this check and instead, always checks for the current user when mounted.

The code in question originated in #4258. @Amplifiyer may have input on what edge-case I'm missing.

The crux of the issue is really the race-condition that exists: https://github.com/aws-amplify/amplify-js/milestone/23.  If the hosted UI flow completed before `Auth.currentAuthenticatedUser()` resolved, then leaking `amplify-signin-with-hostedUI`to the UI components wouldn't be necessary.  (Note that customers with fully-custom UIs don't have logic for `amplify-signin-with-hostedUI`!)

### Before

> bug-ui-react

### After

> fix-ui-react

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx [hunk 1]
```diff
@@ -62,9 +61,7 @@ export class AmplifyAuthenticator {
     });
 
     appendToCognitoUserAgent('amplify-authenticator');
-    const byHostedUI = localStorage.getItem(SIGNING_IN_WITH_HOSTEDUI_KEY);
-    localStorage.removeItem(SIGNING_IN_WITH_HOSTEDUI_KEY);
```
Comment (ashika01) [inline packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx:66]: Doesn't removing this impact us? If I remember from react code, one of the reasons it was added was to check if the redirect is actually from the hostedUI?!

**Comment by Amplifiyer** [comment]
>  If the hosted UI flow completed before `Auth.currentAuthenticatedUser()` resolved, then leaking `amplify-signin-with-hostedUI`to the UI components wouldn't be necessary.

This is the exact reason why this flag exist in the `aws-amplify-react` package. Do we know why it's different in `ui-components`?

**Comment by ericclemmons** [comment]
@Amplifiyer The UI code was copy/pasted from `aws-amplify-react`, so this behavior may exist there. I'll pop open a sample & check it out.

File: packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx (hunk 1 above)
Comment (ericclemmons) [reply packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx:66]: How do you mean impact? My understanding is the same as yours, but with my research (and videos), the refresh behavior blocked the app from rendering at all.

Here's where it originates:
https://github.com/aws-amplify/amplify-js/blob/8cf409559754aa7dd41d42b7a024845a6226709e/packages/auth/src/Auth.ts#L1816-L1818

My take is that there's a leaky abstraction (or incorrect separation of concerns), where `Auth.ts` is setting a value that **only our UI components** are responsible for removing.

I discovered this while working on #5676, but we may address this differently (especially if the race-condition is fixed!) in the near future.

**Comment by ericclemmons** [comment]
Going to update to match #4258 (using the `amplify-redirected-from-hosted-ui`)

**Pull request merged**